src/gui/addnewtorrentdialog.cpp

```diff
@@ -621,32 +636,63 @@ void AddNewTorrentDialog::displayContentTreeMenu(const QPoint &)
     }
 }
 
+void AddNewTorrentDialog::setAddTorrentParams(const BitTorrent::AddTorrentParams &inParams)
+{
+    torrentParams = inParams;
+    
+    if (!torrentParams.savePath.isEmpty())
+        setSavePath(torrentParams.savePath);
+    
+    if (!torrentParams.category.isEmpty()) {
+        int index = ui->categoryComboBox->findText(torrentParams.category);
+        
+        if (index != -1) {
+            ui->categoryComboBox->setCurrentIndex(index);
+        }
+        else {
+            ui->categoryComboBox->addItem(torrentParams.category);
```

Comment:
And what if the category is absent? Say, user mistype it in the command line? I would prefer a program that tells me about that.